client/src/Components/Cart.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useContext, useEffect } from "react";
+import { CartContext } from "./CartContext";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import styles from "../Styles/Cart.module.css";
+import axios from "axios";
+import CustomerInfoForm from "./CustomerInfoForm";
+import { addDays } from "date-fns";
+
+// react_app_baseURL is env variable provided by heroku on hosting
+axios.defaults.baseURL =
+  process.env.REACT_APP_baseURL || "http://localhost:4000";
+
+const Cart = () => {
+  const [cart] = useContext(CartContext);
+  const [disableBtn, setDisableBtn] = useState(true)
+  const [order, setOrder] = useState([]);
+  const [startDate, setStartDate] = useState(null);
+  const [customerInfo, setCustomerInfo] = useState({
+    name: '',
+    email: ''
+  });
+  // payload will be used in fetch request to send order information to server
+  const payload = { ...customerInfo, items: order, pickupDate: startDate };
+
+  const handleInput = (e) => {
+    setCustomerInfo({
+      ...customerInfo,
+      [e.target.name]: e.target.value,
+    });
+  };
+  // disable checkout button until all fields are flled in
+  useEffect(()=>{
+    if (customerInfo && startDate) {
+      setDisableBtn(false)
```

Comment:
this doesn't actually work as intended because the `customerInfo` object is always truthy. What you want is to check each key value

Reply:
fixed